pkg/catalog/traffictarget.go

```diff
@@ -179,25 +176,8 @@ func (mc *MeshCatalog) getTCPRouteMatchesFromTrafficTarget(trafficTarget smiAcce
 			return nil, errNoTrafficSpecFoundForTrafficPolicy
 		}
 
-		// TODO(#1521): Create an actual TCP route match once v1alpha4 TCPRoute spec is available
-		// Hack: derive ports from 'ports' label on the TCPRoute resource
-		portsStr, ok := tcpRoute.Labels["ports"]
-		if !ok {
-			continue
-		}
-		portStrList := strings.Split(portsStr, ",")
-		var ports []int
-		for i := range portStrList {
-			portStrList[i] = strings.TrimSpace(portStrList[i])
-			port, err := strconv.Atoi(portStrList[i])
-			if err != nil {
-				return nil, errors.Errorf("Invalid port value %s, must be an integer", portStrList[i])
-			}
-			ports = append(ports, port)
-		}
-
 		tcpRouteMatch := trafficpolicy.TCPRouteMatch{
-			Ports: ports,
+			Ports: tcpRoute.Spec.Matches.Ports,
```

Comment:
I suspect we could run into a situation where Matches is empty and this could throw a nil pointer dereference.
Should we check if we have Matches and ports first?

Reply:
matches and ports are already required as part of the CRD https://github.com/servicemeshinterface/smi-sdk-go/blob/master/crds/specs.yaml#L108-L114

Reply:
Matches is a struct and not a pointer, so we won't have a nil pointer dereference.
```
// TCPRouteSpec is the specification of a TCPRoute
type TCPRouteSpec struct {
	// Route match for inbound traffic
	Matches TCPMatch `json:"matches,omitempty"`
}
```

Reply:
Thanks for answering my question!